### PR TITLE
add aten.topk implementation

### PR DESCRIPTION
**Pull request by lanluo-nvidia**
# Description

Add aten.topk implementation

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: tests/py/dynamo/conversion/test_topk_aten.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import torch.nn as nn
+from parameterized import parameterized
+from torch.testing._internal.common_utils import run_tests
+
+from .harness import DispatchTestCase
+
+
+class TestSortConverter(DispatchTestCase):
+    @parameterized.expand(
+        [
```
Comment (apbose) [inline tests/py/dynamo/conversion/test_topk_aten.py:11]: Can you add more test cases with 2D and 3D cases? Also cases on how it would handle duplicate values.
Also could you once print the decomposed graph to see if aten.topk.value is invoked or not.

File: py/torch_tensorrt/dynamo/conversion/impl/topk.py [hunk 2]
```diff
@@ -134,3 +134,38 @@ def sort(
         return topk_layer.get_output(0), topk_layer.get_output(1)
     else:
         return topk_layer.get_output(0)
+
+
+def topk(
```
Comment (apbose) [inline py/torch_tensorrt/dynamo/conversion/impl/topk.py:139]: Hi can you consolidate the topk() and the sort() function, something like the sort() function calls the topk function with the k arg as the dim?

Also in the below the sorted argument should also be handled to return the elements in sorted order.

File: py/torch_tensorrt/dynamo/conversion/impl/topk.py [hunk 3]
```diff
@@ -103,31 +103,38 @@ def argmin(
     )
 
 
-def sort(
+def topk(
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/conversion/impl/topk.py:106]: Make a sort function that just calls topk with the right parameters.

File: tests/py/dynamo/conversion/test_topk_aten.py
Comment (apbose) [inline tests/py/dynamo/conversion/test_topk_aten.py:1]: The function structure as discussed last time looks good to me.
In the above you mentioned that topk in TRT returns sorted by default. So was curious if there was a mismatch in the test cases where sorted=False.
Also could you add a validator to the topk function such that k values greater than 3840 will not be supported by TRT layer. Something like this - https://github.com/pytorch/TensorRT/blob/main/py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py#L264-L269

Comment (lanluo-nvidia) [reply tests/py/dynamo/conversion/test_topk_aten.py:1]: added the topk validator, was having difficulties try to get the input tensor shape will discuss this offline with @narendasan for help.

**Review by apbose — COMMENTED** [review]
Looks good to me! Pending on CI.
